### PR TITLE
Prefer https:// for URLs throughout project

### PR DESCRIPTION
**Pull request by jdufresne**

### CODE REVIEW

**Comment by kennethreitz** [comment]
Did you verify all of these work?

**Comment by jdufresne** [comment]
Hi Kenneth,

I've done this same change on a number of different repositories now. I automated this change using the script:

https://github.com/jdufresne/commit-messages/blob/master/scripts/checkhttp.py

At a high level, this script does the following:

1. Scans all text files for strings that look like a URL starting with `http://`
2. Tries fetching the same URL but with `https://`
3. Check the final destination is a 200 response with protocol `https://` (to avoid redirects back to `http://`)
4. If it looks good, replace all occurrences

After running the script, I analyze the results locally using a visual diff tool (I like [meld](https://wiki.gnome.org/Apps/Meld)). I undo the changes for tests that look to be intentionally using an `http://` URL (requests has a lot of these). I also try to remove `en/latest/` paths from readthedocs that didn't previously exist. I manually test a bunch of URLs especially any that look suspicious with a wildly different path. I re-analyze the results using GitHub's diff tool (I find looking at a diff in multiple ways can help emphasize different differences).

Many of the changed URLs share a domain. So after spot checking a few from these domains, we can be confident with the rest.

So, in summary, a script has confirmed that all these changes result in `https://` URLs that 200 and several have been manually tested.

Please feel free to re-run the script to verify or do additional manual testing.

File: docs/user/advanced.rst [hunk 1]
```diff
@@ -25,8 +25,8 @@ Let's persist some cookies across requests::
 
     s = requests.Session()
 
-    s.get('http://httpbin.org/cookies/set/sessioncookie/123456789')
```
Comment (nateprewitt) [inline docs/user/advanced.rst:28]: Why was this removed?

File: docs/user/advanced.rst [hunk 2]
```diff
@@ -69,7 +69,7 @@ If you want to manually add cookies to your session, use the
 Sessions can also be used as context managers::
 
     with requests.Session() as s:
-        s.get('http://httpbin.org/cookies/set/sessioncookie/123456789')
```
Comment (nateprewitt) [inline docs/user/advanced.rst:72]: Same question here.

**Comment by jdufresne** [comment]
Thanks for catching those mistakes. Restored and fixed in the latest revision.
